src/GitHub.Exports/Models/RepositoryModelBase.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System;
+using System.Diagnostics;
+using System.IO;
+using GitHub.Extensions;
+using GitHub.Primitives;
+using GitHub.Services;
+using GitHub.UI;
+
+namespace GitHub.Models
+{
+    /// <summary>
+    /// The base class for local and remote repository models.
+    /// </summary>
+    [DebuggerDisplay("{DebuggerDisplay,nq}")]
+    public class RepositoryModelBase : NotificationAwareObject, IRepositoryModel
+    {
+        UriString cloneUrl;
+        Octicon icon;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RepositoryModelBase"/> class.
+        /// </summary>
+        /// <param name="name">The repository name.</param>
+        /// <param name="cloneUrl">The repository's clone URL.</param>
+        protected RepositoryModelBase(
+            string name,
+            UriString cloneUrl)
+        {
+            Guard.ArgumentNotEmptyString(name, nameof(name));
+            Guard.ArgumentNotNull(cloneUrl, nameof(cloneUrl));
+
+            Name = name;
+            CloneUrl = cloneUrl;
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RepositoryModelBase"/> class.
+        /// </summary>
+        /// <param name="path">
+        /// The path to the local repository from which repository name and clone URL will be
+        /// extracted.
+        /// </param>
+        protected RepositoryModelBase(string path)
+        {
+            Guard.ArgumentNotNull(path, nameof(path));
+
+            var dir = new DirectoryInfo(path);
+            if (!dir.Exists)
+                throw new ArgumentException("Path does not exist", nameof(path));
+            var uri = GitService.GitServiceHelper.GetUri(path);
+            Name = uri?.RepositoryName ?? dir.Name;
+            CloneUrl = GitService.GitServiceHelper.GetUri(path);
```

Comment:
If `CloneUrl` is null here (and it can be), then the `Owner` getter will blow up below. We probably need to be aware/handle the case of loading a git repository that doesn't have a remote set by returning null on the `Owner` below in this case (probably good enough?)